addons/cluster-autoscaler/v1.4.0.yaml

```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler
+  labels:
+    k8s-app: cluster-autoscaler
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: cluster-autoscaler
+  template:
+    metadata:
+      labels:
+        k8s-app: cluster-autoscaler
+    spec:
+        containers:
+          - image:
+            name: {{IMAGE}}
+            resources:
+              limits:
+                cpu: 100m
+                memory: 300Mi
+              requests:
+                cpu: 100m
+                memory: 300Mi
+            command:
+              - ./cluster-autoscaler
+              - --cloud-provider={{CLOUD_PROVIDER}}
+              - --nodes={{MIN_NODES}}:{{MAX_NODES}}:{{GROUP_NAME}}
+            env:
+              - name: AWS_REGION
```

Comment:
My org has been using it ever since https://github.com/kubernetes/contrib/pull/1377 was merged, works very well for us. I've also seen some discussions around it in #sig-aws also, most people seem to have no problems with it. The biggest caveat is that there's no official image released yet (to my knowledge) on `gcr.io/google_containers` so you have to build/push your own image from master.